Oney threshold front check

GhDj opened the pull request:
- [x] Bump PayPlug version with a 4th digit (ex: 1.1.0 -> 1.1.0.1)
- [x] Generate payplug_woocommerce.zip
- [x] Install payplug_woocommerce.zip on your Woocommerce environment
- [x] Login to your newly installed PayPlug plugin
- [x] Validate a simple payment with PayPlug
- [x] Validate a refund partial/total with PayPlug
- [x] Check apache logs

issam-payplug reviewed (changes requested):
Line 53 & 54 will cause a problem in the BO...

You should use the new  properties of the Gateway classess ($this->oney_thresholds_min and $this->oney_thresholds_max from 331) instead of ($this->min_oney_price and $this->max_oney_price) 
in the BO (_PayplugGateway.php_) befaore saving settings I have to compare the new min max with min_oney_price and max_oney_price and if their not within the threshold I get an error (ex: **"Le montant doit être compris entre 100€ et 3000€."**) but now when they are overriten in  _PayplugGatewayOney3x.php_ I ill get arbitrary error (**"Le montant doit être compris entre 1203€ et 2474€."**), which is wrong.


Conclusion: make it so that in check_oney_is_available() you will do the validation with **$this->oney_thresholds_min** and **$this->oney_thresholds_max** instead of **$this->min_oney_price** and **$this->min_oney_price**.

issam-payplug commented:
2) amount_check() method in PayplugWoocommerceHelper class is  unnecessary
3) in any Gateway class that inherit PayplugGateway the new MIN MAX values already selected from database in the properties **$this-oney_thresholds_min** and  **$this->oney_thresholds_max** (in PayplugGateway.php constructor)... So it is unnecessary to use **PayplugWoocommerceHelper::get_local_oney_threshold()** in PayplugGatewayOney3x.php  because it will bring the same values again from the database and rerun the same sql request again.
4) get_minimum_amount() and get_minimum_amount() from PayplugWoocommerceHelper can be removed all together because they're only used in PayplugGateway@validate_order_amount() where the MIN and Max are already selected from the database as mentioned above (then validate_order_amount() can use **$this-oney_thresholds_min** and  **$this->oney_thresholds_max** instead)

GhDj commented:
1 - I didn't see any BO error.
2 - Yes, used when I wanted to implement the front code seperated from the back code, which what we should consider during the refactoring.
3 - PayplugWoocommerceHelper::get_local_oney_threshold() still necessary, even for future use.
4 - get_minimum_amount() and get_minimum_amount()  where already there, but refactoring them was an obligation to avoid getting always the account values (100 and 3000) , but I refactored and deleted them.
5 - We should consider testing this before sqving qny vqlues of thresholds ...

PPmmesquita commented on src/Gateway/PayplugGatewayOney3x.php:
```diff
@@ -50,8 +50,9 @@ public function __construct()
 
-		$this->min_oney_price = PayplugWoocommerceHelper::get_local_oney_threshold()['oney_thresholds_min'];
-		$this->max_oney_price = PayplugWoocommerceHelper::get_local_oney_threshold()['oney_thresholds_max'];
+		/*$this->min_oney_price = PayplugWoocommerceHelper::get_local_oney_threshold()['oney_thresholds_min'];
```
this should be deleted
